### PR TITLE
stats: fix UB in from_slice

### PR DESCRIPTION
**Pull request by ordian**
As documented in https://doc.rust-lang.org/std/mem/union.MaybeUninit.html, 
```rust
let x: i32 = unsafe { MaybeUninit::uninit().assume_init() }; // undefined behavior! ⚠️
```

### CODE REVIEW

File: src/stats.rs [hunk 1]
```diff
@@ -93,40 +93,56 @@ fn value_histogram_index(size: u32) -> Option<usize> {
...
+where
+	F: Fn(&mut Cursor<&[u8]>) -> T,
+{
+	// SAFETY:
+	// https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#initializing-an-array-element-by-element
+	let mut data: [MaybeUninit<T>; N] = unsafe { MaybeUninit::uninit().assume_init() };
+	for item in &mut data[..] {
+		item.write(reader(cursor));
+	}
+	data.map(|x| unsafe { x.assume_init() })
```
Comment (ordian) [inline src/stats.rs:107]: this copies the data unfortunately
using transmute currently not possible with const generics: https://github.com/rust-lang/rust/issues/61956
but I can change it to a macro instead

Comment (ordian) [reply src/stats.rs:107]: The following functions are unstable, but could be used in the future:
* https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#method.uninit_array
* https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#method.array_assume_init

Comment (arkpar) [reply src/stats.rs:107]: Might as well get rid of unsafe here.
```
 let plain: [u32; 1024] = [Default::default(); 1024];
 let mut atomic: [AtomicU32; 1024] = a.map(AtomicU32::new);
 // Read into `atomic` below.
```
Shoudl be optimized away as well.

Comment (ordian) [reply src/stats.rs:107]: Not sure it's optimized this way though. The produced code seems different: https://godbolt.org/z/rGsKPnrnf
Also to sanity, create a bench:
```
from_slice/Unsafe       time:   [1.0590 µs 1.0613 µs 1.0636 µs]                               
from_slice/Safe           time:   [2.1197 µs 2.1255 µs 2.1326 µs]  
```

Comment (arkpar) [reply src/stats.rs:107]: Ok, let's leave it as it is then.

**Pull request merged**